server/ethereum/ethereum.test.js

```diff
@@ -77,6 +142,35 @@ test('watcher pins file from registerData function', async done => {
     })
 })
 
+test('firing a listen event adds a new contract to listen to into state', async done => {
```

Comment:
Should we write another test to make sure that duplicate smart contracts aren't added? I understand that's a test in the `state` tests, but we're not specifically making sure that the listener handler is firing the `smartContracts.add` function, so it might be nice to have that redundancy here. 

In a perfect world, using spies and some robust testing framework, this might test: `listen event fires smartcontract.add function`. That way we wouldn't have to do these tests twice. I'll leave this one up to you though, since i'm confident behavior is working as we want it

Reply:
how would you test that?